python/src/main/java/org/apache/zeppelin/python/IPythonInterpreter.java

```
@@ -150,33 +154,27 @@ public boolean checkIPythonPrerequisite() {
       Process proc = processBuilder.start();
       int ret = proc.waitFor();
       if (ret != 0) {
-        LOGGER.warn("Fail to run pip freeze.\n" +
-            IOUtils.toString(new FileInputStream(stderrFile)));
-        return false;
+        return "Fail to run pip freeze.\n" +
+            IOUtils.toString(new FileInputStream(stderrFile));
       }
       String freezeOutput = IOUtils.toString(new FileInputStream(stdoutFile));
       if (!freezeOutput.contains("jupyter-client=")) {
-        InterpreterContext.get().out.write("jupyter-client is not installed\n".getBytes());
-        return false;
+        return "jupyter-client is not installed.";
       }
       if (!freezeOutput.contains("ipykernel=")) {
-        InterpreterContext.get().out.write("ipkernel is not installed\n".getBytes());
-        return false;
+        return "ipkernel is not installed";
       }
       if (!freezeOutput.contains("ipython=")) {
-        InterpreterContext.get().out.write("ipython is not installed\n".getBytes());
-        return false;
+        return "ipython is not installed";
       }
       if (!freezeOutput.contains("grpcio=")) {
-        InterpreterContext.get().out.write("grpcio is not installed\n".getBytes());
-        return false;
+        return "grpcio is not installed";
       }
       LOGGER.info("IPython prerequisite is meet");
-      return true;
     } catch (Exception e) {
       LOGGER.warn("Fail to checkIPythonPrerequisite", e);
-      return false;
```

if there's an exception we should return a generic error string here?

---

Fixed